crates/forge/bin/cmd/test/mod.rs

```diff
@@ -814,6 +827,49 @@ fn persist_run_failures(config: &Config, outcome: &TestOutcome) {
     }
 }
 
+/// Generate test report in JUnit XML report format.
+fn junit_xml_report(results: &BTreeMap<String, SuiteResult>, verbosity: u8) -> Report {
+    let mut junit_report = Report::new("Test run");
+    junit_report.set_timestamp(Utc::now());
+    for (suite_name, suite_result) in results {
+        let mut test_suite = TestSuite::new(suite_name);
+        test_suite.set_system_out(suite_result.summary());
+        for (test_name, test_result) in &suite_result.test_results {
+            let test_status = if test_result.status.is_failure() {
+                let mut status = TestCaseStatus::non_success(NonSuccessKind::Failure);
+                status.set_message(test_result.reason.clone().unwrap_or_default());
+                status
```

Comment:
not necessary for this PR, but we could track the kind of failure and set the failure type https://docs.rs/quick-junit/latest/quick_junit/enum.TestCaseStatus.html